app/models/appointment_reminder.rb

```diff
@@ -14,4 +15,45 @@ class AppointmentReminder < ApplicationRecord
     sent: "sent",
     cancelled: "cancelled"
   }, _prefix: true
+
+  scope :due_today, -> { where(remind_on: Date.current, status: [:pending]) }
+
+  def localized_message
+    I18n.t(
+      message,
+      facility_name: appointment.facility.name,
+      patient_name: patient.full_name,
+      appointment_date: appointment.scheduled_date,
+      locale: patient_locale(patient)
+    )
+  end
+
+  def next_communication_type
+    if preferred_communication_method && !previously_communicated_by?(preferred_communication_method)
+      return preferred_communication_method
+    end
+    unless previously_communicated_by?(backup_communication_method)
+      return backup_communication_method
+    end
+    nil
+  end
+
+  private
+
+  def previously_communicated_by?(method)
+    communications.any? { |communication| communication.communication_type == method }
+  end
+
+  def preferred_communication_method
+    CountryConfig.current[:name] == "India" ? "missed_visit_whatsapp_reminder" : nil
+  end
+
+  def backup_communication_method
+    "missed_visit_sms_reminder"
+  end
+
+  # this is temporary. will be replaced by facility address localization
```

Comment:
Can you explain more when this will go away? Or link up the clubhouse story in the comment with the follow on work?

Reply:
I'll link the ticket. It's a subsequent ticket in this epic and I released a draft of it a while ago.